### PR TITLE
Various cleanups

### PR DESCRIPTION
**Pull request by awemany**
This is a pretty random collection of minor changes, some of which are fixes of my own mistakes. If wanted, I can squash them to fewer ones, of course.

### CODE REVIEW

**Comment by awemany** [comment]
981e0e4 is to prevent a failure like https://api.travis-ci.org/v3/job/401387928/log.txt

File: src/thinblock.cpp [hunk 1]
```diff
@@ -1500,7 +1500,7 @@ void AddThinBlockInFlight(CNode *pfrom, uint256 hash)
         std::pair<uint256, CNode::CThinBlockInFlight>(hash, CNode::CThinBlockInFlight()));
 }
 
-void SendXThinBlock(CBlockRef pblock, CNode *pfrom, const CInv &inv)
+void SendXThinBlock(ConstCBlockRef pblock, CNode *pfrom, const CInv &inv)
```
Comment (sickpig) [inline src/thinblock.cpp:1503]: this is working cuase `const` reference could be bind to an non `const` object right?

**Review by sickpig — APPROVED** [review]
approved, I just have a minor question related to `SendXThinBlock`

File: src/thinblock.cpp (hunk 1 above)
Comment (awemany) [reply src/thinblock.cpp:1503]: You mean when passing in a `CBlockRef`? Yes, to the best of my knowledge :D

**Pull request merged**